support/sier-codec/src/lib.rs

```diff
@@ -48,14 +50,76 @@ impl Parser {
     pub fn struct_def(&self, name: &str) -> Option<&Arc<StructDef>> {
         self.structs.values().find(|s| s.type_name() == name)
     }
-}
 
-impl Default for Parser {
-    fn default() -> Self {
-        Parser {
-            structs: HashMap::new(),
+    fn transform_serde_value<'a>(
```

Comment:
As discussed in our VC, these should be free functions (without a self parameter).

Reply:
Also in a separate file. json.rs or something similar.